File: target_rotatedsortedarray.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        start=0
        end=len(nums)-1
        while(start<=end):
            mid=start+(end-start)//2
            if nums[mid]==target:
                return mid
            if nums[start]<=nums[mid]:
                if target<nums[mid] and target>=nums[start]:
                    end=mid-1
                else:
                    start=mid+1
            else:
                if target>nums[mid] and target<=nums[end]:
                    start=mid+1
                else:
                    end=mid-1
        return -1

File: TargetRotatedSortedArray.java
class TargetRotatedSortedArray 
{
    int right;
    int left;
    int mid;
    public int search(int[] nums, int target) 
    { 
     int first=0;
     int last = nums.length-1;
        
     while(first<last)
     {
      mid=first+(last-first)/2;
      if(nums[mid]>nums[last])
          first=mid+1;
      else
          last=mid;
     }   
        
     if(target>=nums[first]&&target<=nums[nums.length-1])
     {
         left = first;
         right=nums.length-1;
     }
     else
     {
     left=0;
     right=first-1;    
     }
        
     while(left<=right)
     {
     mid=left+(right-left)/2;
     if(nums[mid]==target) 
         return mid;
     else if(target>nums[mid])
         left=mid+1;
     else
         right=mid-1;
     }
        return -1;        

        
    }
    
}